src/game/game.js

```diff
@@ -422,8 +422,7 @@
                 get() {
 
                     try {
-                        runCodeCache[userId].memory._parsed = JSON.parse(runCodeCache[userId].memory.get() || "{}");
-                        runCodeCache[userId].memory._parsed.__proto__ = null;
+                        runCodeCache[userId].memory._parsed = _sandboxedEval('JSON.parse')(runCodeCache[userId].memory.get() || "{}");
```

Comment:
So, I'm failing to see how this makes sense.

When runCode is called, a new closure is created. Every function declared inside that closure, including the get(){} in Memory, is assigned to that closure. Since the Memory property is defined every time, it always gets replaced by a get(){} inside the new closure, with the newest _sandboxedEval created by that function. 

Then inside the same execution of runCode, driver.evalCode is called, running the main.loop function. When it returns, user code has finished executing (or it should, leaky sandbox makes this tricky). So the Memory property shouldn't be accessed by anything anymore.

So, maybe I'm not understanding something, but it seems to me that _sandboxedEval is trapped inside the runCode closure, and no other user will ever see that same value.

Am I making sense? Did I miss something?